glib/src/subclass/types.rs

```diff
@@ -741,6 +741,15 @@ unsafe extern "C" fn instance_init<T: ObjectSubclass>(
     let private_offset = (*data.as_mut()).private_offset;
     let ptr = obj as *mut u8;
     let priv_ptr = ptr.offset(private_offset);
+
+    assert!(
+        priv_ptr as usize & (mem::align_of::<PrivateStruct<T>>() - 1) == 0,
+        "Private instance data has higher alignment requirements ({}) than \
+         the allocation from GLib. If alignment of more than 16 bytes \
```

Comment:
It's 32 bytes on 64 bit platforms but if someone relies on that things will likely fail on 32 bit platforms so let's only mention the lower value here.